crates/sui-move-build/tests/linter/redundant_custom.exp

```diff
@@ -0,0 +1,52 @@
+
+warning[Lint W01001]: non-composable transfer to sender
+   ┌─ tests/linter/redundant_custom.move:14:9
+   │
+13 │     public fun custom_transfer_bad(o: S1, ctx: &mut TxContext) {
+   │                ------------------- Returning an object from a function, allows a caller to use the object and enables composability via programmable transactions.
+14 │         transfer::transfer(o, tx_context::sender(ctx))
+   │         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+   │         │                     │
+   │         │                     Transaction sender address coming from here
+   │         Transfer of an object to transaction sender address in function custom_transfer_bad
+
+warning[Lint W01001]: potentially unenforceable custom transfer/share/freeze policy
+   ┌─ tests/linter/redundant_custom.move:17:16
+   │
+17 │     public fun custom_share_bad(o: S1) {
+   │                ^^^^^^^^^^^^^^^^ - An instance of a module-private type with a store ability to be shared coming from here
+   │                │                 
+   │                Potential unintended implementation of a custom share_object function.
+18 │         transfer::share_object(o)
+   │                   ------------ Instances of a type with a store ability can be shared using the public_share_object function which often negates the intent of enforcing custom share policy (implemented through calling the private share_object function variant in the module defining this type)
+
+warning[Lint W01001]: possible owned object share
+   ┌─ tests/linter/redundant_custom.move:18:9
+   │
+17 │     public fun custom_share_bad(o: S1) {
+   │                                 - A potentially owned object coming from here
+18 │         transfer::share_object(o)
+   │         ^^^^^^^^^^^^^^^^^^^^^^^^^
+   │         │                      │
+   │         │                      Creating a fresh object and sharing it within the same function will ensure this does not abort.
+   │         Potential abort from a (potentially) owned object created by a different transaction.
+
+warning[Lint W01001]: potentially unenforceable custom transfer/share/freeze policy
+   ┌─ tests/linter/redundant_custom.move:21:16
+   │
+21 │     public fun custom_freeze_bad(o: S1) {
+   │                ^^^^^^^^^^^^^^^^^ - An instance of a module-private type with a store ability to be frozen coming from here
+   │                │                  
+   │                Potential unintended implementation of a custom freeze_object function.
+22 │         transfer::freeze_object(o)
+   │                   ------------- Instances of a type with a store ability can be frozen using the public_freeze_object function which often negates the intent of enforcing custom freeze policy (implemented through calling the private freeze_object function variant in the module defining this type)
```

Comment:
The length of this message makes me wonder if we should move this into a help message on this warning, and have a shorter error message for this?

Reply:
I will leave it up to @tnowacki to decide. It's a tough tradeoff here between the succinctness and the desire for the message to be descriptive enough...

Reply:
Totally agreed. It's a fine line (and actually incredibly difficult IMO) to determine these types of things. 

My two cents are this should most likely become a help message eventually, BUT if we don't already really use "help" in our error messages elsewhere it's probably best to not start now and keep this as-is -- and make adding in help messages and the like a separate bit of work.

Reply:
We do have a "note" section in diagnostics. So it might be worthwhile to move a lot of that text into a note.

Reply:
I think we don't quite have any block of text quite this large in other diagnostics. I have used a note for some edge case / ambiguous behavior for local variable copy/move. 
This feels appropriate here too (maybe)

Reply:
Gave it a shot!